rmw_fastrtps_dynamic_cpp/include/rmw_fastrtps_dynamic_cpp/TypeSupport_impl.hpp

```diff
@@ -693,27 +550,32 @@ size_t TypeSupport<MembersType>::getEstimatedSerializedSize(
           if (!member->is_array_) {
             current_alignment += getEstimatedSerializedSize(sub_members, field, current_alignment);
           } else {
-            void * subros_message = nullptr;
             size_t array_size = 0;
-            size_t sub_members_size = sub_members->size_of_;
-            size_t max_align = calculateMaxAlign(sub_members);
 
             if (member->array_size_ && !member->is_upper_bound_) {
-              subros_message = field;
               array_size = member->array_size_;
             } else {
-              array_size = get_array_size_and_assign_field(
-                member, field, subros_message, sub_members_size, max_align);
+              if (!member->size_function) {
+                RMW_SET_ERROR_MSG("unexpected error: size function is null");
+                return false;
+              }
+              array_size = member->size_function(field);
```

Comment:
Unrelated note: I'm not sure why `getEstimatedSerializedSize` is needed here and not in `rmw_cyclonedds`

Reply:
The calculation of the sample serialization size is only done for unbounded types (see [here](https://github.com/ros2/rmw_fastrtps/blob/ccb15906683c65911ac632d16360f36edd0d02ac/rmw_fastrtps_dynamic_cpp/include/rmw_fastrtps_dynamic_cpp/TypeSupport_impl.hpp#L928)). For bounded types, the maximum serialized size calculated when constructing the type support object (for instance, [here](https://github.com/ros2/rmw_fastrtps/blob/ccb15906683c65911ac632d16360f36edd0d02ac/rmw_fastrtps_dynamic_cpp/include/rmw_fastrtps_dynamic_cpp/MessageTypeSupport_impl.hpp#L58)) is used.

Knowing the payload size of the sample is needed in order to know the size of the payload that needs to be allocated.

On `rmw_cyclonedds` it is done [here](https://github.com/ros2/rmw_cyclonedds/blob/91aed377a0a2e39664d0eae401ee49e9c419540a/rmw_cyclonedds_cpp/src/serdata.cpp#L210)